streams/test-utils/src/main/java/org/apache/kafka/streams/test/OutputVerifier.java

```diff
@@ -238,4 +239,117 @@
         compareKeyValueTimestamp(record, expectedRecord.key(), expectedRecord.value(), expectedRecord.timestamp());
     }
 
+    /**
+     * Compares a {@link ProducerRecord} with the provided value and headers and throws an {@link AssertionError} if
+     * the {@code ProducerRecord}'s value or headers is not equal to the expected value or headers.
+     *
+     * @param record a output {@code ProducerRecord} for verification
+     * @param expectedValue the expected value of the {@code ProducerRecord}
+     * @param expectedHeaders the expected headers of the {@code ProducerRecord}
+     * @param <K> the key type
+     * @param <V> the value type
+     * @throws AssertionError if {@code ProducerRecord}'s value or headers is not equal to {@code expectedValue} or {@code expectedHeaders}
+     */
+    public static <K, V> void compareValueHeaders(final ProducerRecord<K, V> record,
+                                                  final V expectedValue,
+                                                  final Headers expectedHeaders) throws AssertionError {
+        Objects.requireNonNull(record);
+
+        final V recordValue = record.value();
+        final Headers recordHeaders = record.headers();
+        final AssertionError error = new AssertionError("Expected <" + expectedValue + "> with headers=" + expectedHeaders +
+                "but was <" + recordValue + "> with headers=" + recordHeaders);
+
+        if (recordValue != null) {
+            if (!recordValue.equals(expectedValue)) {
+                throw error;
+            }
+        } else if (expectedHeaders != null) {
+            throw error;
+        }
+
+        if (recordHeaders != expectedHeaders) {
```

Comment:
for headers, we need to do the same as for `recordValue` ? (I think you c&p from `compareValueTimestamp` -- there it's different because timestamp is a `long` and cannot be `null`.

Reply:
ack.